### PR TITLE
feat: add VPC ID and Subnet IDs patch

### PR DESCRIPTION
**Pull request by dkoshkin**
Fixes https://github.com/d2iq-labs/capi-runtime-extensions/issues/201
Fixes https://github.com/d2iq-labs/capi-runtime-extensions/issues/200

Depends on https://github.com/d2iq-labs/capi-runtime-extensions/pull/219

Tested manually (at-least that the values are set in `AWSCluster`)
```
apiVersion: cluster.x-k8s.io/v1beta1
kind: Cluster
metadata:
  name: <NAME>
spec:
  topology:
    variables:
      - name: clusterConfig
        value:
          aws:
            network:
              vpc:
                id: vpc-1234567890
              subnets:
                - id: subnet-1
                - id: subnet-2
                - id: subnet-3
```

This is what the `AWSCluster` looked like:
```
  spec:
    network:
      subnets:
      - id: subnet-1
        isPublic: false
      - id: subnet-2
        isPublic: false
      - id: subnet-3
        isPublic: false
      vpc:
        availabilityZoneSelection: Ordered
        availabilityZoneUsageLimit: 3
        id: vpc-1234567890
    region: us-west-2
```

### CODE REVIEW

File: docs/content/customization/aws/network.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          aws:
+            network:
+              vpc:
+                id: vpc-1234567890
```
Comment (supershal) [inline docs/content/customization/aws/network.md:45]: are we plan to support all the vpcSpec properties in future? https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta1/network_types.go#L184

File: pkg/handlers/aws/mutation/network/inject.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package network
+
+import (
+	"context"
+	_ "embed"
```
Comment (supershal) [inline pkg/handlers/aws/mutation/network/inject.go:8]:
```suggestion
	
```

**Review by supershal — APPROVED** [review]
LGTM. except a `nit`

File: docs/content/customization/aws/network.md (hunk 1 above)
Comment (dkoshkin) [reply docs/content/customization/aws/network.md:45]: Yeah I can see us adding support for things like the CIDR

**Pull request merged**